packages/nextjs/app/page.tsx

```diff
@@ -2,48 +2,105 @@
 
 import Link from "next/link";
 import type { NextPage } from "next";
-import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
+import { useContractRead } from "wagmi";
+import { BugAntIcon, MagnifyingGlassIcon, RocketLaunchIcon, UserGroupIcon } from "@heroicons/react/24/outline";
+import BuildersGrid from "~~/components/batch/BuildersGrid";
+import deployedContracts from "~~/contracts/deployedContracts";
+
+const BATCH_REGISTRY_ADDRESS = "0x65E42251A2ca0906c640331afd77f45F4682c56C";
+const BATCH_REGISTRY_ABI = deployedContracts[31337].BatchRegistry.abi;
 
 const Home: NextPage = () => {
+  const { data: checkedInCount, isLoading } = useContractRead({
+    address: BATCH_REGISTRY_ADDRESS,
+    abi: BATCH_REGISTRY_ABI,
+    functionName: "checkedInCounter",
+  });
```

Comment:
I think you should use `useScaffoldReadContract`, it's way easier to use and would not require the contract address and ABI.

Reply:
Okay, thanks I will do that. Thank you